app/decorators/user_decorator.rb

```diff
@@ -1,9 +1,9 @@
 class UserDecorator < ApplicationDecorator
   delegate_all
 
-  def proposal_url(proposal)
+  def proposal_url(proposal, force_public_url=false)
     event = proposal.event
-    if model.staff_for? event
+    if model.staff_for?(event) && !force_public_url
```

Comment:
I've just gone over this again and I don't see why not. My memory is that proposal_url was being used in a few different spots besides #create_for, and looking around now I don't see that being the case.